ql/cashflows/cpicoupon.cpp

```diff
@@ -207,6 +207,15 @@ namespace QuantLib {
             return notional() * (I1 / I0);
     }
 
+    Real CPICashFlow::indexFixing() const {
+        if (observationDate_ != Date()) {
+            return CPI::laggedFixing(cpiIndex(), observationDate_, observationLag_, interpolation_);
+        } else {
+            // we get to this branch when the deprecated constructor was used; it will be phased out
+            return CPI::laggedFixing(cpiIndex(), fixingDate() + observationLag_, observationLag_,
```

Comment:
(a) I'm not sure that this is the same as the `else` branch in `amount`, and (b) I'd try to avoid duplication.  What do you think about moving the code in `amount` here and calling `indexFixing` in `amount`?

Reply:
sure things

Reply:
Good idea, will change it accordingly